packages/help-center/src/components/help-center-content.tsx

```diff
@@ -76,6 +76,17 @@ const HelpCenterContent: React.FC< { isRelative?: boolean; currentRoute?: string
 
 	const isUserElegible = data?.eligibility.is_user_eligible ?? false;
 
+	useEffect( () => {
+		// Prevent not eligible users from accessing wapuu.
+		if ( ! isUserElegible && ! isLoadingEnvironment ) {
+			recordTracksEvent( 'calypso_helpcenter_redirect_not_eligible_user_to_homepage', {
+				pathname: window.location.pathname,
+				search: window.location.search,
+			} );
+			navigate( '/' );
+		}
+	}, [ isUserElegible, isLoadingEnvironment, navigate ] );
```

Comment:
I would remove the navigate function from the dependencies array. It happened to me frequently that the navigate function is not persistent across renders and might trigger many re renders automatically :)

Reply:
🤔 I think it is necessary since it is a dependency of that function.
I have tried without it, despite the ESLint error, it did not work well.

Reply:
My bad then, I had pretty bad experiences in the past with navigate function in the deps array :)